### ISSUE TITLE
Dynamic publications page

### ISSUE THREAD

**Issue by crisaless** [informatics-isi-edu/atlas-d2k-www#28]
The publications page should update from the listings of publications from Chaise:

Source in Chaise:
https://dev.rebuildingakidney.org/chaise/recordset/#2/Common:Publication

Instructions are in README.md (see attached ZIP)

This is an example script I was given, which generates the bib file (see attached ZIP):

`python3 -m atlas_d2k.cli.publication.export2bib --host dev.rebuildingakidney.org --consortium GUDMAP --from-year 2021 --to-year 2021`

Goals: 
- Publish results for consortium = RBK and GUDMAP at /publications/index.html
- Publish results for consortium = RBK at /rebuildingakidney/publications/index.html
- Publish results for consortium = GUDMAP at /gudmap/publications/index.html

The format should be similar to what we have currently - groups publications by year and shows by date in descending order.

Also add a tag in the HTML output for each publication at the very end that includes the consortium - much like it shows in recordset here: https://www.atlas-d2k.org/chaise/recordset/#2/Common:Publication. 

[publications script.zip](https://github.com/informatics-isi-edu/atlas-d2k-www/files/12164236/publications.script.zip)

**Comment by ladukaraniket** [comment]
@crisaless @hongsudt 

# Following approaches can be taken to achieve this

## Python Script followed by a jekyll build
- The above mentioned python script can be used to generate the bib files for GUDMAP and RBK respectively
- A ```jekyll build``` can be created. This would then generate all the respective static pages and create list of publications sorted by year.

> ### Pros
> - simple and easy to implement
> - We currently do not use paging on the publications page. Adding support for paging would be easier if we decide to do so in future.

> ### Cons
> - Routinely need to create a build to update the publications.
> - The site might not reflect the latest data at all times and updates would  depend on the build cycle.

<hr>

## Using JavaScript to dynamically fetch real time data and append the HTML to DOM
- Data can be queried using HTTPS request and the DOM can then be updated to reflect the latest data.

> ### Pros
> - The site would reflect the latest data at all times.

> ### Cons
> - Implementation is comparatively more time consuming.
> - Implementing paging in future would be trickier.

**Comment by hongsudt** [comment]
What's the design? Do you plan to have a drop-down that can toggle between different years, or different tabs for different years?

**Comment by crisaless** [comment]
The design should be like it is displayed on https://www.rebuildingakidney.org/publications/ - all publications listed by month/year in descending order, with Headings for each year.

**Comment by ladukaraniket** [comment]
Yes, that's how the design would look. 
We just need to finalize the approach that needs to be taken.

**Comment by crisaless** [comment]
There's one other approach I want to explore. Instead of building HTML pages to represent the publications, we could just send users to the Chaise listing of publications. Especially since this allows users to filter the list and see any connections to other data. 

But I would still want a dynamic method for using JavaScript to dynamically fetch real time data (the second approach listed above) but just to grab the x number of most recent entries for: 

- a) all of atlas (Publications list filtered for RBK and GUDMAP) 
- b) just RBK publications  and 
- c) just GUDMAP publications.

I'll be meeting with NIDDK program staff on 9/5 to get clarity on whether they think we should be listing particulars per consortium (ie, news and publications). But at the very least, I will want to be able to list the 4 or 5 most recent entries for all of atlas to appear on the homepage in a list with the following HTML (most recent publication first):

```
<ul class="posts">
  <li><a href="[DOI]">[Title]</a> <small>([Publication Venue], [Year], [Consortium])</small></li>
  <li><a href="[DOI]">[Title]</a> <small>([Publication Venue], [Year], [Consortium])</small></li>
  <li><a href="[DOI]">[Title]</a> <small>([Publication Venue], [Year], [Consortium])</small></li>
  <li><a href="[DOI]">[Title]</a> <small>([Publication Venue], [Year], [Consortium])</small></li>
</ul>
```

**Comment by crisaless** [comment]
@hongsudt This last comment is the new "needs" for representing publications. We'll just use Chaise as the destination but I want to be able to display the most recent posts on the jekyll site where needed. Does this need any further discussion before I ask Aniket to implement this?

**Comment by hongsudt** [comment]
Yes, please. Let's discuss the use cases. I thought we would delay working
on this dynamic until after site migration..
